testsuite/src/main/scala/endpoints/testsuite/BasicAuthTestApi.scala

```diff
@@ -12,4 +13,13 @@ trait BasicAuthTestApi extends algebra.Endpoints with algebra.BasicAuthenticatio
     textResponse
   )
 
+  val protectedEndpointViaBuilder =
+    anEndpoint
+      .withMethod(Get)
+      .withUrl(path / "users")
+      .withTextResponse
+      .withBasicAuth
```

Comment:
@ShaneDelmore Just to clarify, Julien's example would work correctly. To imitate this wrong behaviour, one would have to use `endpoint` instead of `authenticatedEndpoint`, like this:
```
endpoint(request(method, url, requestEntity, basicAuthentication),response)
```
For comparison, `authenticatedEndpoint` is defined like so:
```
endpoint(request(method, url, requestEntity, basicAuthentication), authenticated(response))
```

So, in current API basic auth is safe, because it gives you guarantee that both, response and request, will be authenticated(so, request contains/parse `Authorisation` header and response contains/parse 200/403 status code. With the new API this guarantee is gone, because user can override part of the mechanism. 

I'm experimenting with ways of fixing it, but all the tries either removes composability of builders or complicates the API in extraordinary way.